File: src/main/java/nl/waterjeloen/quoridor/Player.java
package nl.waterjeloen.quoridor;

import java.util.function.Function;

public class Player {
    private final String name;
    private Location location;
    private int walls;
    private final Function<Location, Boolean> wins;

    public Player(String name, Location location, Function<Location, Boolean> wins) {
        this.name = name;
        this.location = location;
        this.walls = 10;
        this.wins = wins;
    }

    public String getName() {
        return name;
    }

    public Location getLocation() {
        return location;
    }

    public void changeLocation(Location location) {
        this.location = location;
    }

    public void removeWall() {
        --walls;
    }

    public boolean hasWalls() {
        return walls > 0;
    }

    public int getWalls() {
        return walls;
    }

    public String toString() {
        return name;
    }

    public boolean hasWon() { return wins.apply(location); }
}

File: src/main/scala/nl/zwaaltjes/quoridor/Player.scala
package nl.zwaaltjes.quoridor

final case class Player(name: String, private var playerLocation: Location, private var playerWalls: Int = 10) {
  def location: Location =
    playerLocation

  def changeLocation(newLocation: Location): Unit =
    playerLocation = newLocation

  def walls: Int =
    playerWalls

  def hasWalls: Boolean =
    playerWalls != 0

  def useWall(): Unit =
    playerWalls -= 1

  override def toString: String =
    name
}